internal/controller/clusterinstance/rendered_object.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterinstance
+
+import (
+	"encoding/json"
+	"sort"
+	"strconv"
+
+	"golang.org/x/exp/maps"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type action string
+
+const (
+	actionRender   action = "render"
+	actionSuppress action = "suppress"
+	actionPrune    action = "prune"
+)
+
+type RenderedObject struct {
+	object unstructured.Unstructured
+	action action
+}
+
+type syncWaveMap struct {
+	data map[int][]RenderedObject
+}
+
+type RenderedObjectCollection struct {
+	prune, suppress, render syncWaveMap
+}
+
+func (r *RenderedObject) SetObject(manifest map[string]interface{}) error {
+	// Marshal the input manifest to JSON
+	if content, err := json.Marshal(manifest); err != nil {
+		return err
+	} else if err = json.Unmarshal(content, &r.object); err != nil {
```

Comment:
using `else if` in this way is a bit strange. Couldn't these be two separate blocks entirely at the cost of one extra line of code?